#!/usr/bin/env python
PACKAGE = "ros_plane"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# Toggle linear/circular path
gen.add("FLAG", bool_t, 0, "Toggle linear/circular path", True)

# Desired Velocity
gen.add("VA_D", double_t, 0, "Desired Speed", 35, 0, 100)

# line origin 
r = gen.add_group("r")
r.add("RN", double_t, 0, "Line origin north component", 0, -1, 1)
r.add("RE", double_t, 0, "Line origin east component", 0, -1, 1)
r.add("RD", double_t, -100, "Line origin down component", -100, 0, -500)

# line direction
q = gen.add_group("q")
q.add("QN", double_t, 0, "Line direction north component", -0.5, -1, 1)
q.add("QE", double_t, 0, "Line direction east component", -1.0, -1, 1)
q.add("QD", double_t, 0, "Line direction down component", -0.05, -1, 1)

# orbit center
c = gen.add_group("c")
c.add("CN", double_t, 0, "Orbit center north component", 1.0)
c.add("CE", double_t, 0, "Orbit center east component", 1.0)
c.add("CD", double_t, 0, "Orbit center down component", -100.0)

# orbit radius
gen.add("RHO", double_t, 0, "Orbit radius", 400)

# direction of orbit
gen.add("LAMBDA", int_t, 0, "Direction of orbit", -1, -1, 1)

exit(gen.generate(PACKAGE, "ros_plane", "Manager"))
